### DISCUSSION TITLE
Accessing static from endpoint

### DISCUSSION THREAD

**Discussion by MauritsWilke** [sveltejs/kit#4355] [Q&A]
Hi all!

I have an endpoint that uses images from the `static` folder and was wondering if there is a better practice for this.
My current solution for the path of the image is as follows:
```js
const prefix = process.env.NODE_ENV === "development" ? "http://localhost:3000" : "https://minecraftpfp.com";
// and then for the images 
const image = `${prefix}/image.png`;
```

Is there a better way of doing this? any built-in features I am overlooking? 
Thanks in advance!

**Comment by vekunz** [comment]
It should be possible to drop the domain part completely. If you just use `/image.png`, the browser automatically uses the current domain.

**Reply by MauritsWilke** [reply]
This sadly doesn't work because the code is ran on the server and not in the browser.

**Reply by vekunz** [reply]
So you need to load the file in the backend?

**Reply by MauritsWilke** [reply]
yep

**Comment by stephane-vanraes** [comment]
To be clear, the issue here is making a file (json, image, ...) available both in the assets so users can do `mydomain.com/data.json` and read the data as well as having this data available in endpoints.

The issue is though that while in development you could do ```js
import data from '../../../static/data.json';
``` this is not possible anymore in a reliable way once deployed. 

A solution achieved on Discord is as the first post says twofold:
- construct a path the file
- use fetch to read the file

```js
export async function get({ url }) {
  const data = await fetch(url.origin + '/data.json');
  // other stuff here
}
```

This does _feel wrong_ though, there must a better way.

An alternative approach would be to **not** place the file in static but place it in 'lib/data' so we can do ```js
import data from '$lib/data/data.json';
``` solving the issue for the endpoint.

To now make the file available directly, we simply add an extra endpoibt that does just that:

```js
import data from '$lib/data/data.json';
export async function get() {
  return {
   body: data,
   headers: {
     "Content-Type": "application/json"
   }
}
```
